lib/hanami/events/handler_runner.rb

```diff
@@ -9,7 +9,9 @@ def initialize(handler, logger)
       end
 
       def call(payload)
-        instance_exec(payload, &@handler)
+        return instance_exec(payload, &@handler) if @handler.is_a?(Proc)
+
+        @handler.call(payload)
```

Comment:
okay, I understand. You can drop this line and write something like this:
```ruby
event_bus.subscribe(event_name) { |payload| self.new.call(payload) }
```

[here](https://github.com/hanami/events/pull/33/files#diff-b11209558a2fda36c077dd99714a9eb5R10)